quickwit/quickwit-indexing/src/actors/merge_pipeline.rs

```diff
@@ -418,6 +415,36 @@ impl MergePipeline {
         }
         Ok(())
     }
+
+    async fn fetch_immature_splits(
+        &mut self,
+        ctx: &ActorContext<Self>,
+    ) -> MetastoreResult<Vec<quickwit_metastore::SplitMetadata>> {
+        if let Some(immature_splits) = self.initial_immature_splits_opt.take() {
+            return Ok(immature_splits);
+        }
+        let index_uid = self.params.pipeline_id.index_uid.clone();
+        let node_id = self.params.pipeline_id.node_id.clone();
+        let list_splits_query = ListSplitsQuery::for_index(index_uid)
```

Comment:
it does not change much tbh, but I wonder if the code after the first if statement should have been calling the other (batched) `fetch_immature_splits` function.